09-intermediate-rails/ar-serializers.md

```diff
@@ -0,0 +1,86 @@
+# ActiveModel Serializers
+
+## Learning Goals
+- What is serialization?
+- Why would you use a Serializer?
+- Apply a Serializer to a model
+
+## Introduction
+### What is serialization?
+**Serialization** is the process of translating data structures into a format that can be transmitted.
+
+We have already utilized serialization in Rails without being fully aware of it. Rails is built to do some of this for us by default, but we have the opportunity to customize the way our data is serialized for even more control over our JSON responses.
+
+### API Application
+Let's recall what a controller method looked like in our ada-pets example:
+```ruby
+def index
+  pets = Pet.all
+  render json:  pets.as_json(only: [:id, :name, :age, :human]), status: :ok
+end
+```
+
+We use the `as_json` method to allow us to control exactly the data we want to be returned, rather than returning all of the data that our model provides.
+
+The problem arises in our `show` controller method. What if we want this to return the same fields?
+```ruby
+def show
+  pet = Pet.find_by_id(params[:id])
+  render json: pet.as_json(only: [:id, :name, :age, :human]), status: :ok
+end
+```
+
+We have been programming for a _while_ now, and can see the trouble here is that we've duplicated this code in two different controller methods. So let's come up with a plan to DRY things up!
+
+We will create a **Serializer** to manage the JSON that a model object will return. This will allow us to determine what the JSON response will be for a model in only one place, removing the need for individual controller customization.
+
+
+## Setup
+### Remove `as_json`
+We'll start off by removing the `as_json` method in our controller methods so we can ensure that our modifications will have the desired effect.
+
+1. Remove the `as_json` in the `index` and `show` controller actions.
+2. Run the server and hit those endpoints in the browser or Postman to verify that all model fields are returned.
+
+### Gemfile
+Rails and ActiveRecord have thought about this concept a lot and therefore have provided us with a great gem to help us out.
```

Comment:
Very minor point, I don't think that the ActiveModel Serializers project is officially associated with Rails.